newton/_src/solvers/vbd/solver_vbd.py

```
@@ -1927,6 +1944,79 @@ def solve_rigid_body_iteration(self, state_in: State, state_out: State, contacts
+
+        Returns:
+            Tuple of:
+            - body0: wp.array(int32) body index for shape0
+            - body1: wp.array(int32) body index for shape1
+            - point0_world: wp.array(wp.vec3) world-space contact point on body0
+            - point1_world: wp.array(wp.vec3) world-space contact point on body1
+            - force_on_body1: wp.array(wp.vec3) contact force (world) applied to body1
+            - rigid_contact_count: wp.array(int32) length-1 array with active contact count
+        """
+        if contacts is None:
+            raise ValueError("collect_rigid_contact_forces() requires a Contacts instance.")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)